### PR TITLE
majority of style fixes

### PR DESCRIPTION
**Pull request by johngruen**
@ianb this covers the bulk of required style work. I went ahead and made a PR here with a few things left undone in order to get all add-on style changes in.

### CODE REVIEW

File: addon/webextension/onboarding/slides.html [hunk 1]
```diff
@@ -44,9 +44,9 @@ <h1 data-l10n-id="tourHeaderFour"></h1>
 
         <!-- Clickable elements should be buttons for accessibility -->
         <button id="skip" data-l10n-id="tourSkip" tabindex=1>Skip</button>
-        <button id="prev" tabindex=2 data-l10n-label-id="tourPrevious" style="background-image: url('MOZ_EXTENSION/icons/back.svg');"></button>
-        <button id="next" tabindex=3 data-l10n-label-id="tourNext" style="background-image: url('MOZ_EXTENSION/icons/back.svg');"/></button>
-        <button id="done" tabindex=4 data-l10n-label-id="tourDone" style="background-image: url('MOZ_EXTENSION/icons/done.svg')"></button>
```
Comment (johngruen) [inline addon/webextension/onboarding/slides.html]: Not sure why these were inlined, I used a different arrow for the first one so i moved this stuff to Sass

Comment (ianb) [inline addon/webextension/onboarding/slides.html]: As I recall when I wrote this I used characters, and the inlining was something you did on your cleanup pass.  So, I'll call it your prerogative to change it ;)

**Comment by johngruen** [comment]
Fixes  #2701
Fixes #2683 
Fixes #2731
FIxes #2567 (i think)

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "raven-js": "3.14.0",
     "react": "15.4.2",
     "react-dom": "15.4.2",
+    "react-masonry-component": "^5.0.5",
```
Comment (ianb) [inline package.json]: Should be an exact version (no `^`)

File: server/src/pages/shot/share-buttons.js [hunk 3]
```diff
@@ -23,10 +23,9 @@ exports.ShareButton = class ShareButton extends React.Component {
         isExtInstalled={this.props.isExtInstalled}
       />;
     }
+    const active = this.state.display ? "active" : "inactive";
     return <div>
-      <button className="button primary" id="toggle-share" onClick={ this.onClick.bind(this) }>
-        <span className="flex"><img className="share-icon" src={ this.props.staticLink("/static/img/share.svg")} />Share</span>
-      </button>
+      <button className={`button secondary share ${active}`} id="toggle-share" onClick={ this.onClick.bind(this) } alt="Share" />
```
Comment (ianb) [inline server/src/pages/shot/share-buttons.js]: Should be `title` instead of `alt` (or `aria-label` if you don't want it to display as a tooltip)

File: server/src/pages/shot/view.js [hunk 4]
```diff
@@ -308,29 +305,37 @@ class Body extends React.Component {
       renderGetFirefox = renderExtensionNotification = false;
     }
 
+    console.log(this.props);
```
Comment (ianb) [inline server/src/pages/shot/view.js]: debugging print

File: server/src/pages/shotindex/view.js [hunk 5]
```diff
@@ -91,6 +95,13 @@ class Body extends React.Component {
     );
   }
 
+  getShotType(dimensions) {
+    const squareBuffer = 50;
+    if (dimensions.x - squareBuffer > dimensions.y) return "landscape"
+    else if (dimensions.x < dimensions.y - squareBuffer ) return "portrait"
```
Comment (ianb) [inline server/src/pages/shotindex/view.js]: Should have braces:

```
    if (dimensions.x - squareBuffer > dimensions.y) {
      return "landscape";
    } else if (dimensions.x < dimensions.y - squareBuffer) {
      return "portrait";
    }
```

Also semicolons.

File: server/src/pages/shotindex/view.js [hunk 6]
```diff
@@ -91,6 +95,13 @@ class Body extends React.Component {
     );
   }
 
+  getShotType(dimensions) {
+    const squareBuffer = 50;
```
Comment (ianb) [inline server/src/pages/shotindex/view.js]: There should be some comment about what this values means and why it is 50.  Maybe "An image is 'square' if it's within squareBuffer pixels of being square"?

File: server/src/pages/shotindex/view.js [hunk 7]
```diff
@@ -91,6 +95,13 @@ class Body extends React.Component {
     );
   }
 
+  getShotType(dimensions) {
```
Comment (ianb) [inline server/src/pages/shotindex/view.js]: I'd call this getClipType, since the shot is the entire thing, and this is about the clip image.

File: addon/webextension/icons/menu-myshot.svg [hunk 8]
```diff
@@ -1,17 +1 @@
...
-<rect x="20" y="11" class="st0" width="7" height="7"/>
-<rect x="11" y="20" class="st0" width="7" height="7"/>
-<path class="st0" d="M18,11h-6c-0.5,0-1,0.4-1,1v6h7V11z"/>
-<rect x="20" y="29" class="st0" width="7" height="7"/>
-<path class="st0" d="M35,11h-6v7h7v-6C36,11.5,35.6,11,35,11z"/>
-<path class="st0" d="M29,36h6c0.5,0,1-0.4,1-1v-6h-7V36z"/>
-<rect x="20" y="20" class="st0" width="7" height="7"/>
-<rect x="29" y="20" class="st0" width="7" height="7"/>
-</svg>
+<svg width="46" height="46" viewBox="0 0 46 46" xmlns="http://www.w3.org/2000/svg"><title>Page shot / Menu / My shot</title><path d="M11 11.995c0-.55.455-.995.995-.995h23.01c.55 0 .995.455.995.995v23.01c0 .55-.455.995-.995.995h-23.01c-.55 0-.995-.455-.995-.995v-23.01zM11 25v-2h7v2h-7zm9-5h7v-2h-7v2zm9 5h7v-2h-7v2zm-9 4h7v-2h-7v2zm-2-18h2v25h-2V11zm9 0h2v25h-2V11z" fill="#3E3D40" fill-rule="evenodd"/></svg>
```
Comment (pdehaan) [inline addon/webextension/icons/menu-myshot.svg]: Not sure if we care, but the `<title>` in this SVG thing here references "Page shot".

¯\\\_(ツ)\_/¯

Comment (johngruen) [inline addon/webextension/icons/menu-myshot.svg]: i already clobbered ian's commits once, so i'm not changing anything until he says it's okay

File: addon/webextension/selector/ui.js [hunk 9]
```diff
@@ -420,6 +420,13 @@ this.ui = (function() { // eslint-disable-line no-unused-vars
       } else {
         this.el.classList.remove("bottom-selection");
       }
+
+      if (pos.right < 200 ) {
+        this.el.classList.add("left-selection");
+      } else {
+        this.el.classList.remove("left-selection")
```
Comment (pdehaan) [inline addon/webextension/selector/ui.js]: Not sure if this repo enforces it via ESLint, but this is "missing" a semicolon.

File: package.json (hunk 2 above)
Comment (pdehaan) [inline package.json]: @ianb: do we have rules for including semver range operators?
It looks like all the other version numbers are using exact versions.

TL;DR: Drop the `^` here?

File: addon/webextension/selector/ui.js (hunk 9 above)
Comment (ianb) [inline addon/webextension/selector/ui.js]: I think `npm run lint:js` should catch this, haven't run it locally though

File: addon/webextension/icons/menu-myshot.svg (hunk 8 above)
Comment (ianb) [inline addon/webextension/icons/menu-myshot.svg]: Sure... note `svgo` strips it all for deployment.

File: server/src/pages/shot/share-buttons.js (hunk 3 above)
Comment (pdehaan) [inline server/src/pages/shot/share-buttons.js]: Does the `<button>` element support `alt=""` attribute, or only ``?
Should this be `title="Share"`?

File: server/src/pages/shot/share-buttons.js [hunk 10]
```diff
@@ -86,20 +85,21 @@ class ShareButtonPanel extends React.Component {
       className += " share-panel-with-notification";
     }
     return <div id="share-buttons-panel" className={className}>
-      <div className="wrapper row-space">
+      <div className="wrapper row-wrap share-buttons">
         <a title="Share to Facebook wall or message" onClick={ this.onClickShareButton.bind(this, "facebook") } target="_blank" href={ "https://www.facebook.com/sharer/sharer.php?u=" + encodeURIComponent(this.props.shot.viewUrl) }>
-          <img src={ this.props.staticLink("/static/img/share-facebook.svg") } />
+          <img src={ this.props.staticLink("/static/img/btn-fb.svg") } />
```
Comment (pdehaan) [inline server/src/pages/shot/share-buttons.js]: Nit: do we want `alt=""` attributes on these `` tags?
Quick fix, although probably outside the scope of this PR since they weren't there originally either.

Comment (ianb) [inline server/src/pages/shot/share-buttons.js]: Hmm... not sure if the `title` on the anchor covers it?  Maybe not...

File: server/src/pages/shot/view.js [hunk 11]
```diff
@@ -212,7 +212,7 @@ class Body extends React.Component {
         <div>
           <div className="spacer"/>
           If you do nothing,<br/>
-          this shot will be permanently deleted <TimeDiff date={deleteTime} />.
+          this shot will be permanently deleted in <TimeDiff date={deleteTime} />.
```
Comment (pdehaan) [inline server/src/pages/shot/view.js]: Sorry, I'm not very smart, but my eyes are telling me this is going to say something like:

> _"If you do nothing, this shot will be permanantly deleted in very soon."_

I think all other cases it will read fine though.

File: server/src/pages/shot/view.js (hunk 4 above)
Comment (pdehaan) [inline server/src/pages/shot/view.js]: Do we need this `console.log()`?

File: server/src/pages/shot/share-buttons.js (hunk 10 above)
Comment (johngruen) [inline server/src/pages/shot/share-buttons.js]: it does

File: server/src/pages/shot/view.js (hunk 11 above)
Comment (johngruen) [inline server/src/pages/shot/view.js]: changed `very soon` to `a few seconds`

Comment (ianb) [inline server/src/pages/shot/view.js]: Note the TimeDiff stuff is all going to have to be changed quite a bit for localization

Comment (johngruen) [inline server/src/pages/shot/view.js]: yeah, that's kind of why i pulled the verb`in` out so each of those strings is only a unit of time and localizers can do whatever with the antecedent string

File: server/src/pages/shotindex/view.js [hunk 12]
```diff
@@ -125,7 +141,7 @@ class Body extends React.Component {
             {shot.urlDisplay}
           </div>
         </div>
-        <div className="inner-border"/>
+        </div>
```
Comment (pdehaan) [inline server/src/pages/shotindex/view.js]: nit: indenting looks slightly "off" here with addition of new wrapper `<div>`
